### PR TITLE
Fix memory leak (in theory)

### LINKED ISSUE #907: `DbContext`s should be resolved via a factory, not DI
As per [this documentation](https://docs.microsoft.com/en-us/ef/core/dbcontext-configuration/#the-dbcontext-lifetime), a `DbContext` should not be a long-lived object; it represents a unit-of-work, and thusly it's lifetime should reflect that.

Under normal circumstances, this is easy to conform to, however `MediatR`, the library of choice for implementing the mediator pattern does not create scope-per-requests like Remora.Discord does for responder dispatch. 

This would be *fine*, except for the fact that this leaves it up to the IoC container to handle tracking the lifetime of the objects that are created for these requests, which end up outliving the request itself.

This issue has been at least one of the sources of an ongoing, 2-year-long memory leak that's existed for roughly as long as this project has. 

Below is an image of dotMemory's allocation view, showing that the IoC container does not dispose and release the created `GuildContext`s, causing them to pool and balloon memory usage as they're promoted to Gen2, and live for the lifetime of the application.

image

The solution to this issue is to resolve a context via `AddDbContextFactory`, and requesting an instance of a context from the factory.

Instead of a handler looking something like this:

```cs
public class Handler : IRequestHandler<SomeRequest, SomeResult>
{
  private readonly GuildContext _db;
  public Handler(GuildContext db) => _db = db;

  public async Task<SomeResult> Handle(SomeRequest request, CancellationToken ct)
  {
    return _db.Table.FirstOrDefault(t => t.ID = request.ID);
  }
}
```

It would look more along the lines of this:

```cs
public class Handler : IRequestHandler<SomeRequest, SomeResult>
{
  private readonly IDbContextFactory<GuildContext> _factory;
  public Handler(GuildContext factory) => _factory = factory;

  public async Task<SomeResult> Handle(SomeRequest request, CancellationToken ct)
  {
    await using db = await _factory.CreateContextAsync(ct);
    return db.Table.FirstOrDefault(t => t.ID = request.ID);
  }
}
```

In this way, we remove the reliance on runtime container behavior and deterministically release resources (at least, that's what EF Core would want you to think; some issues on the EFC repo imply there are still some issues with query tracking, but this is a good start).

### PR DESCRIPTION
**Pull request by VelvetToroyashi**
Closes #907

### CODE REVIEW

File: src/Silk.Data/MediatR/Guilds/AddGuild.cs [hunk 1]
```diff
@@ -21,12 +21,13 @@ public sealed record Request(Snowflake GuildID, string Prefix) : IRequest<GuildE
     /// </summary>
     internal sealed class Handler : IRequestHandler<Request, GuildEntity>
     {
-        private readonly GuildContext _db;
-        public Handler(GuildContext db) => _db = db;
+        private readonly IDbContextFactory<GuildContext> _dbFactory;
```
Comment (CalebABG) [inline src/Silk.Data/MediatR/Guilds/AddGuild.cs:24]: Injecting factory from constructor was accidentally removed

**Review by CalebABG — APPROVED** [review]
Only other potential change would be in the `PingCommand.cs` where it injects the `DbContext` instead of the factory, but this is completely fine since using the `services.AddDbContextFactory` will also register the context itself.

Just wanted to mention it for consistency sake since I think all other places inject the `DbContextFactory`, but this can be changed after PR if wanted.

File: src/Silk.Data/MediatR/Guilds/AddGuild.cs (hunk 1 above)
Comment (VelvetToroyashi) [reply src/Silk.Data/MediatR/Guilds/AddGuild.cs:24]: Fixed but uncommitted. I'll push my changes now.

**Comment by VelvetToroyashi** [comment]
Short-term shows rather desirable results: 
image

Gonna merge and build this for now, throw it on prod, and see how long it lasts

**Comment by VelvetToroyashi** [comment]
(Prod (as you can imagine) crashed with OOME)

**Pull request merged**